src/axom/inlet/LuaReader.cpp

```diff
@@ -64,114 +54,105 @@ bool LuaReader::parseString(const std::string& luaString)
     SLIC_WARNING("Inlet: Given an empty Lua string to parse.");
     return false;
   }
-
-  m_luaState = luaL_newstate();
-  if (luaL_loadstring(m_luaState, luaString.c_str()) ||
-      lua_pcall(m_luaState, 0, 0, 0))
-  {
-    SLIC_WARNING(fmt::format("Inlet: Given Lua string could not be loaded: {0}",
-                             luaString));
-    m_luaState = nullptr;
-    return false;
-  }
-
+  m_lua.script(luaString);
   return true;
 }
 
 // TODO allow alternate delimiter at sidre level
 #define SCOPE_DELIMITER '/'
 
-bool LuaReader::findVariable(const std::string& id)
+bool LuaReader::getBool(const std::string& id, bool& value)
 {
-  if (!m_luaState)
-  {
-    SLIC_WARNING(
-      "Lua state is not initialized. Call LuaReader::parseString or LuaReader::parseFile first!");
-    return false;
-  }
+  return getValue(id, value);
+}
 
-  std::string temp_id = id;
-  //TODO: support multiple roots?
-  if (axom::utilities::string::startsWith(temp_id, SCOPE_DELIMITER))
-  {
-    temp_id.erase(0, 1);
-  }
 
-  if (axom::utilities::string::endsWith(id, SCOPE_DELIMITER))
-  {
-    SLIC_WARNING(fmt::format("Variable cannot end with scope delimiter: {0}",
-                             id));
-    return false;
-  }
+bool LuaReader::getDouble(const std::string& id, double& value)
+{
+  return getValue(id, value);
+}
 
-  bool atGlobalScope = true;
-  std::vector<std::string> tokens;
-  axom::utilities::string::split(tokens, temp_id, SCOPE_DELIMITER);
 
-  // Clear the lua stack because we always call with fully qualified names
-  lua_settop(m_luaState, 0);
-  for (std::string token : tokens)
-  {
-    if(atGlobalScope)
-    {
-      lua_getglobal(m_luaState, token.c_str());
-    }
-    else
-    {
-      lua_getfield(m_luaState, -1, token.c_str());
-    }
-    if(lua_isnil(m_luaState, -1))
-    {
-      // variable not found
-      return false;
-    }
-    atGlobalScope = false;
-  }
-
-  return true;
+bool LuaReader::getInt(const std::string& id, int& value)
```

Comment:
Maybe its time to revisit my getInt getBool getDouble getString functions... @kennyweiss  thoughts?

Reply:
Could be -- definitely in a separate PR though.
on the other hand, the current way makes it explicit for the user.